app/scripts/controllers/transactions/index.js

```diff
@@ -753,6 +745,56 @@ export default class TransactionController extends EventEmitter {
     await this.incomingTransactionHelper.update();
   }
 
+  /**
+   * Updates Transaction swap specific properties
+   *
+   * @param {string} txId - The transaction id to be updated
+   * @param {object} swapProperties - Swap specific properties to be updated
+   * @param {string} swapProperties.sourceTokenSymbol
+   * @param {string} swapProperties.destinationTokenSymbol
+   * @param {string} swapProperties.type
+   * @param {string} swapProperties.destinationTokenDecimals
+   * @param {string} swapProperties.destinationTokenAddress
+   * @param {string} swapProperties.swapMetaData
+   * @param {string} swapProperties.swapTokenValue
+   * @param {string} swapProperties.estimatedBaseFee
+   * @param {string} swapProperties.approvalTxId
+   */
+  updateTransactionSwapProperties(
```

Comment:
This method added in order to skip using a bigger one (`updateTransaction`) as we plan to make specific updates.